src/connectors/answers/__tests__/connectAnswers-test.ts

```diff
@@ -0,0 +1,447 @@
+        hits: hitsWithPosition,
+        isLoading: false,
+      }),
+      false
+    );
+
+    // wait
+    await wait(300);
+    // but no more rendering
+    expect(renderFn).toHaveBeenCalledTimes(4);
+  });
+
+  describe('getRenderState', () => {
+    it('returns the render state', () => {
+      const { instantSearchInstance, widget, helper } = setupTestEnvironment({
+        hits: [],
+      });
+
+      expect(
+        widget.getRenderState(
+          {},
+          createInitOptions({
+            instantSearchInstance,
+            state: helper.state,
+            helper,
+          })
+        )
+      ).toEqual({
+        answers: {
+          hits: [],
+          isLoading: false,
+          widgetParams: {
+            queryLanguages: ['en'],
+            attributesForPrediction: ['description'],
+          },
+        },
+      });
+
+      widget.init!(
+        createInitOptions({
+          instantSearchInstance,
+          state: helper.state,
+          helper,
+        })
+      );
+
+      expect(
+        widget.getRenderState(
+          {},
+          createInitOptions({
+            instantSearchInstance,
+            state: helper.state,
+            helper,
+          })
+        )
+      ).toEqual({
+        answers: {
+          hits: [],
+          isLoading: false,
+          widgetParams: {
+            queryLanguages: ['en'],
+            attributesForPrediction: ['description'],
+          },
+        },
+      });
+    });
+  });
+
+  describe('getWidgetRenderState', () => {
+    it('returns the widget render state', async () => {
+      const { instantSearchInstance, widget, helper } = setupTestEnvironment({
+        hits: [{ title: '' }],
+      });
+
+      widget.init!(
+        createInitOptions({
+          instantSearchInstance,
+          state: helper.state,
+          helper,
+        })
+      );
+
+      helper.state.query = 'a';
+      widget.render!(
+        createRenderOptions({
+          instantSearchInstance,
+          state: helper.state,
+          helper,
+        })
+      );
+
+      // render the loading state
+      expect(
+        widget.getWidgetRenderState(
+          createRenderOptions({
+            instantSearchInstance,
+            state: helper.state,
+            helper,
+          })
+        )
+      ).toEqual({
+        hits: [],
+        isLoading: true,
+        widgetParams: {
+          queryLanguages: ['en'],
+          attributesForPrediction: ['description'],
+        },
+      });
+
+      await wait();
+
+      expect(
+        widget.getWidgetRenderState(
+          createRenderOptions({
+            instantSearchInstance,
+            state: helper.state,
+            helper,
+          })
+        )
+      ).toEqual({
+        hits: [{ title: '', __position: 1 }],
+        isLoading: false,
+        widgetParams: {
+          queryLanguages: ['en'],
+          attributesForPrediction: ['description'],
+        },
+      });
```

Comment:
Not sure why but this test throw `Received: serializes to the same string` even though it should work (`.toContainEqual` show that they are equals). This is still an opened issue but this workaround fixes it https://github.com/facebook/jest/issues/8475#issuecomment-537830532

Reply:
I think it might be happening because the hits array has the same items, but not the same properties (like __escaped), is that possible?

Reply:
it could also be due to the circular dependency fixed in #4618

Reply:
I ran the tests after merging master and it still occurs

Reply:
> I think it might be happening because the hits array has the same items, but not the same properties (like __escaped), is that possible?

`__escaped: true` is returned when logging `widget.getWidgetRenderState`

Reply:
It was because of the changes in the connector. I fixed it :) 
https://github.com/algolia/instantsearch.js/pull/4581/commits/51507ec25400f155fc63e16956ffb7e488f5439f